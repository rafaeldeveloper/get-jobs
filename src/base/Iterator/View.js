import React from 'react';
import { Card as CardSemantic } from 'semantic-ui-react';
import  Loading from 'base/Loading';

export default ({ item, refetch, data, ...extraProps }) => {


    refetch();

    return (

        data ?
            <CardSemantic.Group {...extraProps}>
                {
                    data?.map(dataProps => {

                        const adatapetedProps = { ...dataProps }

                        return item(adatapetedProps)

                    }
                    )
                }
            </CardSemantic.Group>
            :

            <Loading />

    )


}

