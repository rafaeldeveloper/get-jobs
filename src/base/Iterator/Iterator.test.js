import React from 'react';
import Iterator from 'base/Iterator';
import renderer from 'react-test-renderer';




test('Iterator renders correctly', () => {

    const component = renderer.create(
        <Iterator/>
    );

    let tree = component.toJSON();
    
    expect(tree).toMatchSnapshot();
''
});

