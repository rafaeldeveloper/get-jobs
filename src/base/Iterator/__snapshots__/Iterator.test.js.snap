// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Iterator renders correctly 1`] = `
<div
  className="ui active transition visible dimmer"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="ui medium inverted text loader"
    >
      Loading
    </div>
  </div>
</div>
`;
